bin/pgenv

```diff
@@ -260,20 +270,283 @@ pgenv_current_version_number(){
+
+        if [ $installed_versions -gt 0 ]; then
+            echo "Cannot delete default configuration while instances installed"
+            echo "If you really want to delete, manually remove [$PGENV_CONFIG_FILE]"
+            exit 1
+        fi
+        # if the file is not here, nothing to do!
+      elif [ ! -f "$PGENV_CONFIG_FILE" ]; then
+            echo "No configuration to delete"
+            exit 1
+    fi
+
+    # remove the file
+    pgenv_debug "Deleting configuration file [$PGENV_CONFIG_FILE]"
+    rm -f "$PGENV_CONFIG_FILE"
+    echo "Configuration file [$PGENV_CONFIG_FILE] deleted"
+}
+
+
+# Utility function to write a single variable in the configuration file.
+#
+# Accepts:
+# - the configuration file name to which write to
+# - the name of the variable
+# - the value of the variable
+# - an optional comment.
+#
+#
+# Invoking it as
+#   pgenv_configuration_write_variable 'pgenv.conf' 'foo' 'bar' 'A foo option'
+#
+# will result in the file in the following:
+#
+#   # A foo option
+#   foo=bar
+#
+# If the variable value is not set, the variable name is placed with a comment sign.
+pgenv_configuration_write_variable(){
+    local file=$1
+    local name=$2
+    local value=$3
+    local comment=$4
+
+    if [ ! -z "comment" ]; then
+        echo "# $comment" >> "$file"
+    fi
+
+    # if no value supplied, put a comment
+    if [ -z "$value" ]; then
+        echo -n "# " >> "$file"
+    fi
+    echo "${name}='${value}'" >> "$file"
+    echo >> "$file"
+}
+
+# Writes the whole current configuration to the current version file
+# or the version file specified. It does a backup copy in the case
+# the file already exists.
+pgenv_configuration_write() {
+    local v=$1
+    CONF=$( pgenv_configuration_file_name $v )
+
+    # sanity check
+    if [ -z "$CONF" ]; then
+        echo "Cannot determine which configuration file to use!"
+        exit 1
+    fi
+
+    # make a backup copy
+    if [ -f "$CONF" ]; then
+        CONF_BACKUP="${CONF}.backup"
+        pgenv_debug "Making backup copy of [$CONF] into [$CONF_BACKUP]"
+        mv "$CONF" "$CONF_BACKUP"
+    fi
+
+    # ok, write the configuration
+    echo "# pgenv configuration created on $(date)" > "$CONF"
+    echo "" >> "$CONF"
+    pgenv_configuration_write_variable "$CONF" "PGENV_DEBUG" "" 'Enables debug output'
+    pgenv_configuration_write_variable "$CONF" "PGENV_STOP_MODE" "$PGENV_STOP_MODE" "Cluster stop mode for 'stop' and 'restart'"
+    pgenv_configuration_write_variable "$CONF" "PGENV_LOG" "$PGENV_LOG" "Path to the cluster log file"
+    pgenv_configuration_write_variable "$CONF" "PGENV_PG_USER" "$PGENV_PG_USER" "Cluster administrator user"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_INITDB" "$PGENV_FLAGS_INITDB" "Initdb flags"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_MAKE" "$PGENV_FLAGS_MAKE" "Make flags"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_CONFIGURE" "$PGENV_FLAGS_CONFIGURE" "Configure flags"
```

Comment:
I think "FLAGS" should come at the end. And maybe use `OPTS`, instead, since they're generally called options rather than flags — at least in English (I think "flags" is more of a Windows word these days). Suggested names:

```
PGENV_INITDB_OPTS
PGENV_MAKE_OPTS
PGENV_CONFIGURE_OPTS
PGENV_START_OPTS
PGENV_STOP_OPTS
PGENV_RESTART_OPTS
```

Reply:
The rationale to name it "flags" is that some commands, like `pg_ctl` do accepts also options (`-o`) and I was wondering if we should keep them separated. The fact that I named "PGENV_FLAGS_XXX" instead of "PGENV_XXX_FLAGS" is that it makes a lot more easier to find all configuration variables at once.
However, I agree to call them OPTS, but we should decide what to do with language options and stop mode, that could be now deprecated in favor of a full single configure variable or pg_ctl variable.